src/OrchardCore/OrchardCore.Media.Abstractions/Events/IMediaEventHandler.cs

```diff
@@ -0,0 +1,11 @@
+namespace OrchardCore.Media.Events
+{
+    public interface IMediaEventHandler
+    {
+        void MediaCreating(MediaCreationContext context);
+        void MediaCreated(MediaCreationContext context);
+        void MediaDeleting(MediaRemoveContext context);
+        void MediaDeleted(MediaRemoveContext context);
+        void MediaDeletedUncomplete(MediaRemoveContext context);
+    }
```

Comment:
@infofromca you keep changing what you're fixing! I'm starting to lose track :)

So events. Good. 
But we need to make a few more changes. Simplify more!

all of these should be async.

We don't need `MediaCreated` or `MediaDeleted` I think - there's nothing you could do to a stream after creating it that I can think of.
what is `MediaDeletedUncomplete`? no need for that either.

What we also need though is `MediaMovingAsync` - for another issue, but as you are making these events please can you add that.

`MediaDeletingContext` not remove.

Reply:
` there's nothing you could do to a stream after creating it that I can think of.`,  I can think of that .
email the result.

MediaDeletedUncomplete did not use in the code, will change to MediaMovingAsync